src/App.vue

```diff
@@ -95,9 +95,13 @@ export default {
   },
   computed: {
     selectedData() {
-      return this.trenchData?.filter((object) => {
-        return object.Type.includes(this.selectedFilter);
-      });
+      if (this.trenchData) {
+        return this.trenchData?.filter((object) => {
+          return object.Type.includes(this.selectedFilter);
+        });
+      } else {
+        return {};
+      }
```

Comment:
@AlZorba side note, the whole function could be rewritten with the `??`. I would personnaly not use it in this case, I find it very difficult to read
```
return this.trenchData?.filter((object) => {
        return object.Type.includes(this.selectedFilter)
} ?? {}
```

"Javascript double question mark is a logical operator that takes two values and returns the right-hand value if the left-hand value is undefined or null, Else returns the left-hand operand."

Reply:
Fancy !